src/main/java/cms/gov/madie/measure/services/TestCaseService.java

```diff
@@ -114,7 +115,19 @@ public TestCase updateTestCase(
     }
 
     // try to persist the Patient Bundle to HAPI FHIR
-    testCase.setHapiOperationOutcome(validateTestCaseJson(testCase, accessToken));
+    if (testCase == null || StringUtils.isBlank(testCase.getJson())) {
+      testCase.setValidResource(false);
+    } else {
+      HapiOperationOutcome hapiOperationOutcome = validateTestCaseJson(testCase, accessToken);
+      testCase.setHapiOperationOutcome(hapiOperationOutcome);
+      if ((hapiOperationOutcome.getCode() >= 200 || hapiOperationOutcome.getCode() <= 299)
+          && hapiOperationOutcome.isSuccessful()) {
+        testCase.setValidResource(true);
+      } else {
+        testCase.setValidResource(false);
+      }
+    }
```

Comment:
From Story AC: Verify if the call to HAPI fails, then the error flag is set to true so the user has to re-validate it later

Looks like the story is taking about errorFlag, here the implementation is about valid resource. The implementation does make sense, but we may have to update story AC so that QA will not have confusion.